AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -501,6 +620,12 @@ private MouseMode CurrentMode
             }
             set
             {
+                if (_currentMode == MouseMode.DragAll && value != MouseMode.DragAll)
+                {
+                    // Finishing a drag all operation, add a NoOp action so later ones do not combine.
+                    actionManager.PerformAction(new NoOpAction());
```

Comment:
This seems like a dirty hack.

Reply:
It is kind of and it could be removed. However, I couldn't think of a more elegant way around the problem that moving an object five steps to the right doesn't isn't done with a single modification to the canvas but five. Combining these into a single action means that undoing makes more sense to the user but you have to make sure you don't combine subsequent moves as well.

You could hide it behind a better interface on the `ActionManager` but ultimately it will be the same thing under the covers.